tests/testthat/test-linear_pool.R

```diff
@@ -411,6 +426,195 @@ test_that("(weighted) quantiles correctly calculated - lognormal family", {
+                         sample_outputs$output_type_id == 3] <-
+    c(150, 300, 500, 350)
+
+  sample_outputs |>
+    dplyr::mutate(horizon = 0, value = 0.75 * value) |>
+    dplyr::bind_rows(sample_outputs) |>
+    dplyr::mutate(output_type_id = paste0(.data[["location"]], .data[["output_type_id"]])) |>
+    linear_pool(
+      weights = NULL,
+      task_id_cols = c("target_date", "target", "horizon", "location"),
+      compound_taskid_set = c("target", "target_date"),
+      n_output_samples = NULL
+    ) |>
+    expect_error(
+      regex = "The specified `compound_taskid_set` is incompatible with ",
+      fixed = TRUE
+    )
+})
+
+
+test_that("Unequal samples across component models for unique of compound task ID set vars throws an error", {
+  # there are four models, "a", "b", "c", and "d".
+  # The first three models each submit 3 samples, while model "d" submits only 1 sample.
+  # We expect an error in this situation, because our methods currently do not support it.
+  sample_outputs <- expand.grid(stringsAsFactors = FALSE,
+                                model_id = letters[1:4],
+                                location = c("222", "888"),
+                                horizon = 1, #week
+                                target = "inc death",
+                                target_date = as.Date("2021-12-25"),
+                                output_type = "sample",
+                                output_type_id = 1:3,
+                                value = NA_real_)
+
+  sample_outputs$value[sample_outputs$location == "222" &
+                         sample_outputs$output_type_id == 1] <-
+    c(40, 30, 45, 80)
+  sample_outputs$value[sample_outputs$location == "222" &
+                         sample_outputs$output_type_id == 2] <-
+    c(60, 40, 75, 20)
+  sample_outputs$value[sample_outputs$location == "222" &
+                         sample_outputs$output_type_id == 3] <-
+    c(10, 70, 15, 50)
+  sample_outputs$value[sample_outputs$location == "888" &
+                         sample_outputs$output_type_id == 1] <-
+    c(100, 325, 400, 300)
+  sample_outputs$value[sample_outputs$location == "888" &
+                         sample_outputs$output_type_id == 2] <-
+    c(250, 350, 500, 250)
+  sample_outputs$value[sample_outputs$location == "888" &
+                         sample_outputs$output_type_id == 3] <-
+    c(150, 300, 500, 350)
+
+  sample_outputs |>
+    dplyr::mutate(horizon = 0, value = 0.75 * value) |>
+    dplyr::bind_rows(sample_outputs) |>
+    dplyr::filter(model_id %in% letters[1:3] | (output_type_id == 1 & location == "222")) |>
+    linear_pool(
+      weights = NULL,
+      task_id_cols = c("target_date", "target", "horizon", "location"),
+      compound_taskid_set = c("target", "location", "target_date"),
+      n_output_samples = NULL
+    ) |>
+    expect_error(
+      regex = "Within each group defined by a combination of the compound task ID set variables",
+      fixed = TRUE
+    )
```

Comment:
This is a matter of personal preference, but I like to put the `expect_error` call at the top level rather than piped at the end.  This reads more naturally to me:
 - rather than "take the sample outputs and mutate them and add them to the existing sample outputs and then do some filtering and then compute a linear pool and then expect an error to have been thrown"
 - i read "expect an error when I call linear pool with these sample outputs"
```suggestion
 sample_outputs <- sample_outputs |>
    dplyr::mutate(horizon = 0, value = 0.75 * value) |>
    dplyr::bind_rows(sample_outputs) |>
    dplyr::filter(model_id %in% letters[1:3] | (output_type_id == 1 & location == "222"))

  expect_error(
    linear_pool(
      weights = NULL,
      task_id_cols = c("target_date", "target", "horizon", "location"),
      compound_taskid_set = c("target", "location", "target_date"),
      n_output_samples = NULL
    ),
  regex = "Within each group defined by a combination of the compound task ID set variables",
  fixed = TRUE
)
```